src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,254 @@
+﻿using System.Collections.Concurrent;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.Net.Security;
+using System.Runtime;
+using Microsoft.Crank.EventSources;
+
+namespace HttpClientBenchmarks;
+
+class Program
+{
+    private static ClientOptions _options = null!;
+    private static List<HttpClient> _httpClients = new();
+    private static ConcurrentBag<long>? _headersTimes;
+    private static ConcurrentBag<long>? _contentStartTimes;
+    private static ConcurrentBag<long>? _contentEndTimes;
+    private static long _successRequests;
+    private static long _badStatusRequests;
+    private static long _exceptionRequests;
+
+    public static async Task<int> Main(string[] args)
+    {
+        var rootCommand = new RootCommand();
+        rootCommand.AddOption(new Option<string>(new string[] { "--url" }, "The server url to request") { Required = true });
+        rootCommand.AddOption(new Option<Version>(new string[] { "--httpVersion" }, "HTTP Version (1.1 or 2.0 or 3.0)") { Required = true });
+        rootCommand.AddOption(new Option<int>(new string[] { "--numberOfClients" }, () => 1, "Number of HttpClients"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--concurrencyPerClient" }, () => 12, "Number of concurrect requests per one HttpClient"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--http11MaxConnectionsPerServer" }, () => 1, "Max number of HTTP/1.1 connections per server"));
+        rootCommand.AddOption(new Option<bool>(new string[] { "--http20EnableMultipleConnections" }, () => false, "Enable multiple HTTP/2.0 connections"));
+        rootCommand.AddOption(new Option<string>(new string[] { "--scenario" }, "Scenario to run") { Required = true });
+        rootCommand.AddOption(new Option<int>(new string[] { "--warmup" }, () => 15, "Duration of the warmup in seconds"));
+        rootCommand.AddOption(new Option<int>(new string[] { "--duration" }, () => 30, "Duration of the test in seconds"));
+
+        rootCommand.Handler = CommandHandler.Create<ClientOptions>(async (options) =>
+        {
+            _options = options;
+            Log("HttpClient benchmark");
+            Log("Options: " + _options);
+
+            Setup();
+            Log("Setup done");
+
+            await Warmup();
```

Comment:
This seems a little weird, because it will cause the client app to basically "pause" briefly between warmup phase and the actual run. I don't think we want that. Rather, I think we want a continuous run where the first N seconds are the warmup phase and the rest is the actual run.